Zend/zend_compile.c

```diff
@@ -1189,22 +1189,40 @@ static zend_string *resolve_class_name(zend_string *name, zend_class_entry *scop
 
 zend_string *zend_type_to_string_resolved(zend_type type, zend_class_entry *scope) {
 	zend_string *str = NULL;
+	uint32_t type_mask = ZEND_TYPE_PURE_MASK(type);
```

Comment:
I really don't know if it is worthwile as I didn't bother with it initially, maybe it is better to just have compatibility for Reflection when it is only ``iterable`` (or ``?iterable``) ?

Reply:
@Girgias IMO it should work for all cases or we should drop it altogether.

Reply:
I'll go back then to how it was initially, as I would think code running for 8.2 would be adapted to handle union types, but that will need to be put in the migration guide anyway.

Reply:
@Girgias Maybe keep the code around, just in case we'll need the BC solution anyway. I think with a BC break this will be a bit more questionable and should at least be mentioned on the mailing list.

Reply:
I'll email internals then